### PR TITLE
feat(sdk-coin-avaxp): implement tx builder for add validator tx

### PR DESCRIPTION
**Pull request by jrafael-simtlix**
Features:
- Build AddValidator Tx 
- Half signed with user key
- Half signed with recovery key
- Full signed
- Get builder tx from raw tx. 
- KeyPair with cb58 

[STLX-16393](https://bitgoinc.atlassian.net/browse/STLX-16393)

### CODE REVIEW

**Review by sskim16 — APPROVED** [review]
LGTM, added documentation / paired with rafa the last few days on these files.

File: modules/sdk-coin-avaxp/src/transaction.ts [hunk 1]
```diff
@@ -48,20 +75,20 @@ export class Transaction extends BaseTransaction {
    * @param {KeyPair} keyPair
    */
   sign(keyPair: KeyPair): void {
-    const keys = keyPair.getKeys();
-    if (!keys.prv) {
+    // getPrivateKey returns type 32 byte, not PrivateKey- which is the right format
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-coin-avaxp/src/transaction.ts:78]: Who is this question addressed to?

File: modules/sdk-coin-avaxp/src/transactionBuilder.ts [hunk 2]
```diff
@@ -9,39 +9,35 @@ import {
...
-  protected _utxos: DecodedUtxoObj[] = [];
-  protected _txFee: BN;
-
-  private _credentials: Credential[]; // how are we passing in multisig?
+  public _signer: KeyPair[] = [];
+  protected recoverSigner = false;
+
+  /**
+   * When using recovery key must be set here
+   * TODO: recovery key signing
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-coin-avaxp/src/transactionBuilder.ts:30]: Can we get a ticket added for this?

File: modules/sdk-coin-avaxp/src/transactionBuilder.ts [hunk 3]
```diff
@@ -91,40 +87,52 @@ export abstract class TransactionBuilder extends BaseTransactionBuilder {
...
-    return this.transaction;
+    throw new NotSupported('from raw transaction is not supported. See TransactionBuilderFactory.from method');
   }
 
+  get hasSigner(): boolean {
+    return this._signer !== undefined && this._signer.length > 0;
+  }
   /** @inheritdoc */
   protected async buildImplementation(): Promise<Transaction> {
+    // TODO: sign with recovery key
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-coin-avaxp/src/transactionBuilder.ts:124]: Same here

File: modules/sdk-coin-avaxp/test/unit/transactionBuilder.ts [hunk 4]
```diff
@@ -24,10 +24,16 @@ describe('AvaxP Transaction Builder', () => {
...
         },
-        (e) => e.message === errorMessage.ERROR_JSON_PARSING
+        (e) => e.message === errorMessage.ERROR_RAW_PARSING
       );
     });
 
+    it('Should validate a correct raw tx', () => {
+      const txBuilder = factory.getTransferBuilder();
+      txBuilder.validateRawTransaction(testData.ADDVALIDATOR_SAMPLES.unsignedTxHex);
+      // should not thorw a error!
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-coin-avaxp/test/unit/transactionBuilder.ts:34]: // should not throw an error

File: modules/sdk-coin-avaxp/src/transaction.ts (hunk 1 above)
Comment (sskim16) [reply modules/sdk-coin-avaxp/src/transaction.ts]: removed, was a comment to myself that i forgot to remove

File: modules/sdk-coin-avaxp/src/transactionBuilder.ts (hunk 2 above)
Comment (sskim16) [reply modules/sdk-coin-avaxp/src/transactionBuilder.ts]: ticket added and updated TODO

File: modules/sdk-coin-avaxp/src/transactionBuilder.ts (hunk 3 above)
Comment (sskim16) [reply modules/sdk-coin-avaxp/src/transactionBuilder.ts]: DITTO

**Review by sskim16 — APPROVED** [review]

**Review by mmcshinsky-bitgo — APPROVED** [review]

**Pull request merged**